packages/theia-integration/src/node/connection-forwarder.ts

```diff
@@ -14,35 +14,30 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 import { Channel, Disposable, DisposableCollection, MessageProvider } from '@theia/core';
-import { Message, MessageReader, MessageWriter } from 'vscode-jsonrpc';
-
-export interface IConnection extends Disposable {
-    readonly reader: MessageReader;
-    readonly writer: MessageWriter;
-    forward(to: IConnection, map?: (message: Message) => Message): void;
-    onClose(callback: () => void): Disposable;
-}
-
-// Temporary fix/workaround to enable comparability with Theia >=1.27 until https://github.com/eclipse-theia/theia/issues/11405 is resolved
-
+import { Socket } from 'net';
+import { createMessageConnection, Message, SocketMessageReader, SocketMessageWriter } from 'vscode-jsonrpc/node';
 /**
  * Forwards messages from service channel to an (raw) `vscode-json-rpc` connection
  */
 export class ConnectionForwarder implements Disposable {
```

Comment:
Should this be renamed to `SocketConnectionForwarder` or `SocketForwarder`? It seems to me that it went from supporting a general `IConnection` to supporting "only" sockets.